management-account/terraform/iam-roles.tf

```
@@ -70,6 +70,50 @@ data "aws_iam_policy_document" "lambda_basic_execution_test" {
     }
   }
 }
+#########################################
+# CoatGithubActionsReadonlyRole
+#########################################
+resource "aws_iam_role"  "coat_github_actions_read_only" {
+  name               = "CoatGithubActionsReadonlyRole"
+  assume_role_policy = data.aws_iam_policy_document.coat_github_actions_read_only.json
+}
+
+data "aws_iam_policy_document" "coat_github_actions_read_only" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    principals {
+      type = "Federated"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:oidc-provider/token.actions.githubusercontent.com"
+      ]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "token.actions.githubusercontent.com:aud"
+      values   = ["sts.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringLike"
+      variable = "token.actions.githubusercontent.com:sub"
+      values   = ["repo:ministryofjustice/operations-engineering:ref:refs/heads/main"]
```

This probably needs updating to 👇 😁 

```suggestion
      values   = ["repo:[ministryofjustice/cloud-optimisation-and-accountability:ref:refs/heads/main"]
```

unless we're building the automation in the older repo for now 😀